planner/core/exhaust_physical_plans.go

```diff
@@ -2003,13 +2003,25 @@ func (p *LogicalProjection) exhaustPhysicalPlans(prop *property.PhysicalProperty
 	if !ok {
 		return nil, true, nil
 	}
-	proj := PhysicalProjection{
-		Exprs:                p.Exprs,
-		CalculateNoDelay:     p.CalculateNoDelay,
-		AvoidColumnEvaluator: p.AvoidColumnEvaluator,
-	}.Init(p.ctx, p.stats.ScaleByExpectCnt(prop.ExpectedCnt), p.blockOffset, newProp)
-	proj.SetSchema(p.schema)
-	return []PhysicalPlan{proj}, true, nil
+	newProps := []*property.PhysicalProperty{newProp}
+	// generate a mpp task candidate if enforced mpp
+	if newProp.TaskTp != property.MppTaskType && p.SCtx().GetSessionVars().IsMPPEnforced() && p.canPushToCop(kv.TiFlash) &&
```

Comment:
It's a very hack code here. Can we find the root cause why properties don't contain `MppTaskType`? These generated plans are dangerous with force MPP type.

Reply:
if projection is the top operator, we should add a option of MPPTaskType. it is not dangenous, because in the physical phase, if it cannot satisfy MPPTaskType, it only generates root tasks.